specs/order_spec.rb

```diff
@@ -76,35 +76,91 @@
       result.must_equal true
     end
   end
+
+  describe "remove_product" do
+    it "Returns true if the product is removed" do
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      order = Grocery::Order.new(1337, products)
+
+      result = order.remove_product("banana")
+      result.must_equal false
+    end
+
+    it "Returns true if the product is removed" do
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      order = Grocery::Order.new(1337, products)
+
+      result = order.remove_product("pears")
+      result.must_equal true
+    end
+  end
+
 end
 
-# TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Order Wave 2" do
+describe "Order Wave 2" do
+  # Arrange
+  orders = Grocery::Order.all
   describe "Order.all" do
     it "Returns an array of all orders" do
-      # TODO: Your test code here!
+      # Action
+      result = orders
+      # # Assertion
+      result.class.must_equal Array
```

Comment:
You should also check that the elements of the array are instances of class `Order`. If it gave you a bunch of strings or something like that, that would be a bug!

```ruby
result.must_be_instance_of Array

# Make sure we got the right number of things from the CSV
result.length.must_equal 100

# Make sure everything has been made into an Order
result.each do |order|
  order.must_be_instance_of Grocery::Order
end
```

Reply:
Thanks for reminding me. I added tests for testing the size of CSV file and each order should be instance of Order.